p2p/discover/udp.go

```diff
@@ -587,10 +602,19 @@ func (req *ping) handle(t *udp, from *net.UDPAddr, fromID NodeID, mac []byte) er
 		ReplyTok:   mac,
 		Expiration: uint64(time.Now().Add(expiration).Unix()),
 	})
-	if !t.handleReply(fromID, pingPacket, req) {
-		// Note: we're ignoring the provided IP address right now
-		go t.bond(true, fromID, from, req.From.TCP)
-	}
+	t.handleReply(fromID, pingPacket, req)
+
+	n := NewNode(fromID, from.IP, uint16(from.Port), req.From.TCP)
+	if time.Since(t.db.lastPingReceived(fromID)) > nodeDBNodeExpiration {
```

Comment:
I think the previous code also did a bond if findnode fails > 0. The new code doesn't seem to do it any more. Are we sure this is correct?

Reply:
This is a good question. In this code path the node can't be unresponsive if it just pinged us.
I think it's better to not rely on findnode failures here because the table's revalidation loop and failure tracking will catch unresponsive nodes later on and throw them out.